src/app/components/UserAssets/Vesting/useListOfUserVestings.ts

```diff
@@ -1,121 +1,91 @@
+              const vestingDetails: DetailsProps = await contractReader.call(
+                'vestingRegistry',
+                'getVestingDetails',
+                [item],
+              );
+
+              // 'type' can be 0 or 1, 0 - Team Vesting, 1 - Vesting
+              // 'typeCreation' can be 1, 2, 3 representing Vesting Registry 1, Vesting Registry 2 and Vesting Registry 3
+
+              const labelTeam =
+                typeCreation[index] === '1' && type[index] === '0'
+                  ? 'team'
+                  : '';
+              const labelGenesys =
+                typeCreation[index] === '1' && type[index] === '1'
+                  ? 'genesis'
+                  : '';
+              const labelOrigin =
+                typeCreation[index] === '2' && type[index] === '1'
+                  ? 'origin'
+                  : '';
+              const labelReward =
+                typeCreation[index] === '3' && type[index] === '1'
+                  ? 'reward'
+                  : '';
+
+              return {
+                asset: Asset.SOV,
+                staking: 'staking',
+                type: labelGenesys || labelTeam || labelReward || labelOrigin,
+                typeCreation: typeCreation[index],
+                cliff: vestingDetails.cliff,
+                duration: vestingDetails.duration,
+                vestingContract: address[index],
+              } as Vesting;
+            }),
+          ).then(result => {
+            setPossibleVestings(result);
+          });
+        }
+      } catch (e) {
+        console.error(e);
+      }
+    }
+    getVestings();
+  }, [vestingsContracts, loadingVestings, account, asset]);
 
   useEffect(() => {
-    if (!account || account === ethGenesisAddress) {
+    if (!account || account === ethGenesisAddress || !possibleVestings.length) {
       setItems([]);
       setLoading(false);
     } else {
       setLoading(true);
-
       const run = async () => {
-        // restrict list to requested asset only if needed
-        const filteredVestings: FullVesting[] = possibleVestings.filter(item =>
-          asset ? item.asset === asset : true,
-        ) as FullVesting[];
-        // get vesting contracts
-        const { returnData: contracts } = await bridgeNetwork.multiCall(
-          Chain.RSK,
-          filteredVestings.map((item, index) => {
-            return {
-              address: getContract(item.registry).address,
-              abi: vestingRegistryAbi,
-              fnName: item.registryMethod,
-              args: [account],
-              key: index.toString(),
-              parser: val => val[0].toString(),
-            };
-          }),
-        );
-
-        return filteredVestings
-          .map((item, index) => {
-            item.vestingContract = contracts[index];
-            return item;
-          })
-          .filter(item => item.vestingContract !== ethGenesisAddress);
+        return possibleVestings as FullVesting[];
       };
```

Comment:
A) Completely remove the run function, as it's unnecessary and will clean up the code below.
B) Do not use **as**. Type casting this way should not be done, since it disables typescript as a safety tool. Typing should be done at the first layer where an object is generated defined, in this case probably useVesting_getVestingsOf.

Reply:
@ShynRou Thanks for reviewing, i added refactoring based on your comments